chore(sinks): add generic to InFlightLimit

fanatid opened the pull request:
Closes #3487.

@bruceg thanks a lot for generic suggestions in the issue!

bruceg commented on src/sinks/util/service.rs:
```diff
@@ -125,11 +125,35 @@ impl<'de> Deserialize<'de> for InFlightLimit {
     }
 }
 
+pub trait InFlightLimitOption {
+    fn unwrap_with(&self, default: &Self) -> Option<usize>;
+}
```
```suggestion
pub trait ParseInFlightLimit {
    fn parse_in_flight_limit(&self, default: &Self) -> Option<usize>;
}
```

fanatid replied:
Just now I realized that instead of custom trait I can use [std::convert::Into](https://doc.rust-lang.org/std/convert/trait.Into.html).

fanatid commented:
> from an `Option<usize>` to an `enum Something { Auto, Fixed(usize) }`

I'm not sure :thinking: in `aws_cloudwatch_logs` `in_flight_limit` value can be number (`InFlightLimit::Fixed`) or nothing (`InFlightLimit::None`), not `auto`.
